promql/engine.go

```diff
@@ -1496,8 +1503,17 @@ func (ev *evaluator) evalVectorSelector(ctx context.Context, vs *parser.VectorSe
 				if ss.Floats == nil {
 					ss.Floats = reuseOrGetFPointSlices(prevSS, numSteps)
 				}
-				ss.Floats = append(ss.Floats, FPoint{F: f, T: ts})
+				val := f
+				if recordOrigT {
+					// This is an info metric, where we want to track the original sample timestamp.
+					val = float64(origT)
+				}
+				ss.Floats = append(ss.Floats, FPoint{F: val, T: ts})
 			} else {
+				if recordOrigT {
+					panic("this should be an info metric, with float samples")
```

Comment:
Use `ev.error`. Need to tell the user what 'this' is.

Reply:
Done, thanks.